parlai/core/worlds.py

```diff
@@ -814,7 +821,9 @@ def batch_observe(self, index, batch_actions, index_acting):
         for i, w in enumerate(self.worlds):
             agents = w.get_agents()
             observation = None
-            if batch_actions[i] is None:
+            if i >= len(batch_actions):
```

Comment:
??

Reply:
hmm i don't remember when i ran into this issue. i'll try removing this

Reply:
ahh yes i remember now. since we batchify offline, the agent does not see that we have a batchsize of e.g. 16 during the final step of validation if  `num_test_exs % 16 != 0`; it'll see something like 4. So it only responds with a list of 4 actions.

I'm adding some logic to the PBatchWorld to handle this instead of this opaque fix here